raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -647,12 +647,13 @@ private boolean handleVoteResponse(
         }
     }
 
-    private int binaryExponentialElectionBackoffMs(int retries) {
+    // visible for testing
+    int binaryExponentialElectionBackoffMs(int retries) {
         if (retries <= 0) {
             throw new IllegalArgumentException("Retries " + retries + " should be larger than zero");
         }
         // upper limit exponential co-efficients at 20 to avoid overflow
-        return Math.min(RETRY_BACKOFF_BASE_MS * random.nextInt(2 << Math.min(20, retries - 1)),
+        return Math.min(RETRY_BACKOFF_BASE_MS * random.nextInt(2 << Math.min(20, retries)),
```

Comment:
Given my understanding of your intention from the PR description:

> That means, when multiple candidates tried to backoff for the 1st retries, they will all get 0ms backoff, and then keep conflicting again.

Is this change enough?

For the 1st retry — considering the call `binaryExponentialElectionBackoffMs(1)` — this changes `randomNextInt(2)` to `randomNextInt(4)` which goes from 50% chance to 25% chance of getting a 0.

Reply:
Yes, you're right. On second thought, I think this change is not necessary. I'll close it.